### PR TITLE
CFP-25694: Move cilium/cilium-cli code into cilium/cilium repository

### LINKED ISSUE #25694: CFP: Move cilium/cilium-cli code into cilium/cilium repository
## Cilium Feature Proposal
        
It's not really a feature, but I propose we move cilium/cilium-cli code into cilium/cilium repository. I'll add a more detailed proposal in [design-cfps](https://github.com/cilium/design-cfps).

edit: an early draft CFP: https://github.com/cilium/design-cfps/pull/9

### PR DESCRIPTION
**Pull request by michi-covalent**
most of commits are from cilium-cli repo. please review these commits:

8b4d060adb - cilium-cli: Remove go.mod
4bd9e9b9a7 - Fix lint errors
55a6ded612 - Remove redundant files
bc6d709368 - Update CODEOWNERS for cilium-cli/ directory
e497a3767c - Build cilium-cli CI Docker image

ignore the following failures:

- `BPF Checks / Check Patch`: it's complaining about commit message format. i'm not going to rewrite commit messages at this point.
- `Build Commits / Check if build works for every commit`: too many commits to build before timing out.

Ref: https://github.com/cilium/design-cfps/pull/9
Fixes #25694

### CODE REVIEW

File: cilium-cli/connectivity/builder/local_redirect_policy.go
Comment (joestringer) [inline cilium-cli/connectivity/builder/local_redirect_policy.go]:
```suggestion
			// Disable until https://github.com/cilium/cilium-cli/issues/2733 is fixed.
```

File: cilium-cli/connectivity/check/deployment.go
Comment (joestringer) [inline cilium-cli/connectivity/check/deployment.go]:
```suggestion
	// https://github.com/cilium/cilium/cilium-cli/issues/361 is resolved.
```

File: cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go
Comment (joestringer) [inline cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go:29]: nit: This change in the go.mod gets reverted immediately after in the next commit, could fold this into the original commit:

```suggestion
		WithCiliumPolicy(clientEgressOnlyDNSPolicyYAML).               // DNS resolution only
```

**Review by joestringer — APPROVED** [review]
Couple of minor suggestions, but overall LGTM. There's a little bit of open question on how to handle the GitHub workflows for the cilium-cli, but I think that's more of a question for how the subsequent cleanup of the cilium/cilium-cli repo goes rather than a question for this PR.

File: cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go
Comment (michi-covalent) [reply cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go:29]: which go.mod change is this? looks like this is pointing a different file 👀

**Review by thorn3r — APPROVED** [review]
Maximum business :chart_with_upwards_trend:. Highlighted commits LGTM

File: cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go
Comment (joestringer) [reply cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go:29]: This line gets changed in the first go.mod commit, then changed back because it broke linter in the next commit. No need to follow up on this.

**Review by joestringer — APPROVED** [review]
Reapproving with the understanding that the trigger workflow changes in the `DONOTMERGE` commit will be removed.

**Review by youngnick — APPROVED** [review]
@michi-covalent  winning the "single largest number of lines in one PR" competition 2024!

LGTM from sig-servicemesh.

**Review by rgo3 — APPROVED** [review]
PR = Big. :rocket:

File: .github/workflows/build-images-ci.yaml
Comment (aanm) [inline .github/workflows/build-images-ci.yaml]: Is this itentional?

File: .github/workflows/build-images-ci.yaml
Comment (aanm) [inline .github/workflows/build-images-ci.yaml]: Small nit, we could add the "platforms" as part of the matrix.include.

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
very nice

**Review by tklauser — APPROVED** [review]
Only one nit about team description in `CODEOWNERS`.

LGTM, thanks a lot for undertaking this effort :heart:

File: .github/workflows/build-images-ci.yaml
Comment (michi-covalent) [reply .github/workflows/build-images-ci.yaml]: oh this is from a temporary commit to test the CI image build: https://github.com/cilium/cilium/pull/34178/commits/e5d1461539526759f3f52f8111d0062425f5be67 i'll remove it before merging 👍

File: .github/workflows/build-images-ci.yaml
Comment (michi-covalent) [reply .github/workflows/build-images-ci.yaml]: done!

**Review by aanm — APPROVED** [review]

**Review by marseel — APPROVED** [review]

**Comment by michi-covalent** [comment]
i'll resolve the conflict and remove the temporary commit 🚀 🙏

**Comment by michi-covalent** [comment]
@dlapcevic ping! cc @cilium/sig-scalability

**Comment by marseel** [comment]
@michi-covalent I think I've already approved on behalf of sig-scalability.

**Pull request merged**
